packages/background/src/keyring/messages.ts

```diff
@@ -662,6 +662,79 @@ export class RequestSignDirectMsg extends Message<{
   }
 }
 
+export class RequestSignEthereumMsg extends Message<{
+  readonly signed: {
+    bodyBytes: Uint8Array;
+    authInfoBytes: Uint8Array;
+    chainId: string;
+    accountNumber: string;
+  };
+  readonly signature: StdSignature;
+}> {
+  public static type() {
+    return "request-sign-ethereum";
+  }
+
+  constructor(
+    public readonly chainId: string,
+    public readonly signer: string,
+    public readonly signDoc: {
+      bodyBytes?: Uint8Array | null;
+      authInfoBytes?: Uint8Array | null;
+      chainId?: string | null;
+      accountNumber?: string | null;
+    },
+    public readonly signOptions: KeplrSignOptions = {}
```

Comment:
It appears this is the same as `RequestSignDirectMsg`, what do you think of just adding `public readonly forEthermint?: boolean` to `RequestSignDirectMsg`?

Reply:
Just pushed a commit with this refactor